src/main/scala/ai/privado/languageEngine/java/tagger/source/IdentifierTagger.scala

```diff
@@ -90,59 +97,55 @@ class IdentifierTagger(cpg: Cpg, ruleCache: RuleCache, taggerCache: TaggerCache)
     memberNameRegex: String,
     ruleInfo: RuleInfo
   ): Unit = {
-    val typeDeclWithMemberNameHavingMemberName = cpg.typeDecl
-      .where(_.member.name(memberNameRegex).filterNot(item => item.name.equals(item.name.toUpperCase)))
-      .map(typeDeclNode => (typeDeclNode, typeDeclNode.member.name(memberNameRegex).l))
+    val typeDeclWithMemberNameHavingMemberName = cachedMember
+      .name(memberNameRegex)
+      .map(memberNode => (memberNode.typeDecl, memberNode))
       .l
     typeDeclWithMemberNameHavingMemberName
-      .distinctBy(_._1.fullName)
       .foreach(typeDeclValEntry => {
-        typeDeclValEntry._2.foreach(typeDeclMember => {
-          val typeDeclVal = typeDeclValEntry._1.fullName
-          // updating cache
-          taggerCache.addItemToTypeDeclMemberCache(typeDeclVal, ruleInfo.id, typeDeclMember)
-          val typeDeclMemberName = typeDeclMember.name
-          // Have started tagging Parameters as well, as in collection points sometimes there is no referencing Identifier present for a local
-          val impactedObjects =
-            cpg.identifier.where(_.typeFullName(typeDeclVal)).l ::: cpg.parameter.where(_.typeFullName(typeDeclVal)).l
-          impactedObjects
-            .whereNot(_.code("this"))
-            .foreach(impactedObject => {
-              if (impactedObject.tag.nameExact(Constants.id).l.isEmpty) {
-                storeForTag(builder, impactedObject, ruleCache)(
-                  InternalTag.OBJECT_OF_SENSITIVE_CLASS_BY_MEMBER_NAME.toString,
-                  ruleInfo.id
-                )
-                storeForTag(builder, impactedObject, ruleCache)(
-                  Constants.id,
-                  Constants.privadoDerived + Constants.underScore + RANDOM_ID_OBJECT_OF_TYPE_DECL_HAVING_MEMBER_NAME
-                )
-                storeForTag(builder, impactedObject, ruleCache)(Constants.catLevelOne, CatLevelOne.DERIVED_SOURCES.name)
-              }
+        val typeDeclMember = typeDeclValEntry._2
+        val typeDeclVal    = typeDeclValEntry._1.fullName
+        // updating cache
+        taggerCache.addItemToTypeDeclMemberCache(typeDeclVal, ruleInfo.id, typeDeclMember)
+        val typeDeclMemberName = typeDeclMember.name
+        // Have started tagging Parameters as well, as in collection points sometimes there is no referencing Identifier present for a local
+
+        (cachedIdentifier.where(_.typeFullName(typeDeclVal)).l ::: cachedParameter.where(_.typeFullName(typeDeclVal)).l)
```

Comment:
Do we need these changes to be done for other languages too? Javascript, Ruby, etc?

Reply:
Not necessarily, only if we need these optimization there. 

We can reflect this to other frontends when we take up the refactoring part to make things common as much as possible, otherwise this is always going to be repeated effort